### PR TITLE
Specify supported OS matrix

### LINKED ISSUE #126679: Error on import PyTorch >= 2.1 on OS X 10.11 with Python 3.10
### 🐛 Describe the bug

Hi, when import `PyTorch` >= 2.1 on OS X 10.11 with Python 3.10 there is an error:

```
Python 3.10.11 (v3.10.11:7d4cc5aa85, Apr  4 2023, 19:05:19) [Clang 13.0.0 (clang-1300.0.29.30)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import torch
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/torch/__init__.py", line 237, in <module>
    from torch._C import *  # noqa: F403
ImportError: dlopen(/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/torch/_C.cpython-310-darwin.so, 2): Symbol not found: __ZNSt19bad_optional_accessD1Ev
  Referenced from: /Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/torch/lib/libc10.dylib (which was built for Mac OS X 10.13)
  Expected in: /usr/lib/libc++.1.dylib
 in /Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/torch/lib/libc10.dylib
```

The latest version of `PyTorch` that works on OS X 10.11 with Python 3.10 seems to be 2.0.1

### Versions

```
Collecting environment information...
PyTorch version: N/A
Is debug build: N/A
CUDA used to build PyTorch: N/A
ROCM used to build PyTorch: N/A

OS: macOS 10.11.6 (x86_64)
GCC version: Could not collect
Clang version: Could not collect
CMake version: Could not collect
Libc version: N/A

Python version: 3.10.11 (v3.10.11:7d4cc5aa85, Apr  4 2023, 19:05:19) [Clang 13.0.0 (clang-1300.0.29.30)] (64-bit runtime)
Python platform: macOS-10.11.6-x86_64-i386-64bit
Is CUDA available: N/A
CUDA runtime version: Could not collect
CUDA_MODULE_LOADING set to: N/A
GPU models and configuration: Could not collect
Nvidia driver version: Could not collect
cuDNN version: Could not collect
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: N/A

CPU:
Intel(R) Core(TM) i7-10750H CPU @ 2.60GHz

Versions of relevant libraries:
[pip3] numpy==1.26.0
[pip3] torch==2.2.2
[conda] Could not collect
```

cc @seemethere @malfet @osalpekar @atalman @svekars @brycebortree @albanD

### PR DESCRIPTION
**Pull request by malfet**
Windows-10 or newer
manylinux-2014
MacOS-11 or newer (but only on Apple Silicon)

Fixes https://github.com/pytorch/pytorch/issues/126679

### CODE REVIEW

**Comment by malfet** [comment]
Not user facing is a weird label here, but from release notes point of view it is not :)

**Review by kit1980 — APPROVED** [review]

File: RELEASE.md [hunk 1]
```diff
@@ -426,6 +427,14 @@ the size restrictions for publishing on PyPI so the default version that is publ
 These special support cases will be handled on a case by case basis and support may be continued if current PyTorch maintainers feel as though there may still be a
 need to support these particular versions of software.
 
+## Operating Systems
+
+PyTorch Linux x86_64 wheels are manylinux2014 compatible, i.e. they should be runnable on any Linux system with glibc-2.17 or above.
+
+Starting from PyTorch-2.3 MacOS binaries are built only for Apple Silicon, and therefore taget MacOS 11 (Big Sur) or newer, but are actively tested against MacOS 14 (Sonoma).
```
Comment (atalman) [inline RELEASE.md:434]: What about aarch64 builds ? Should we mention them here ?

Can we make  it as table ?
OS | Supported Architectures | Details
Linux x86 | CPU, CUDA  | Details about Linux x86
Linux arm64 | CPU, CUDA (Experimental) | Details about Linux arm64 builds 
MacOS x86 | CPU | Starting from 2.3 MacOS is x86 binaries are not officially supported
MacOS arm64 | CPU |
......

File: RELEASE.md [hunk 2]
```diff
@@ -426,6 +427,14 @@ the size restrictions for publishing on PyPI so the default version that is publ
 These special support cases will be handled on a case by case basis and support may be continued if current PyTorch maintainers feel as though there may still be a
 need to support these particular versions of software.
 
+## Operating Systems
+
+PyTorch Linux x86_64 wheels are manylinux2014 compatible, i.e. they should be runnable on any Linux system with glibc-2.17 or above.
```
Comment (atalman) [inline RELEASE.md:432]: Can we make  it as table ?
OS | Supported Architectures | Details
Linux x86 | CPU, CUDA  | Details about Linux x86
Linux arm64 | CPU, CUDA (Experimental) | Details about Linux arm64 builds 
MacOS x86 | CPU | Starting from 2.3 MacOS is x86 binaries are not officially supported
MacOS arm64 | CPU |
......

**Review by huydhn — APPROVED** [review]

File: RELEASE.md (hunk 1 above)
Comment (malfet) [reply RELEASE.md:434]: Let's do it in the followup PR, as I don't know what standard they adhere to right now

File: RELEASE.md (hunk 2 above)
Comment (malfet) [reply RELEASE.md:432]: Sounds good to me, though I would not mention accelerator here, as technically it should have no affect on supported OS matrix.

**Pull request closed without merging**